### PR TITLE
Camila2

### PR DESCRIPTION
**Pull request by ConfettiRaindrop**

### CODE REVIEW

File: app/src/main/java/com/example/myapplication/MatchData.java [hunk 1]
```diff
@@ -59,28 +59,46 @@ public String returnAllData() {
...
+        this.teamNumber = teamNumber;   
+    }
+    
+    public String getMatchNumber() {    
+        return matchNumber;    
+    }
+
+    public void setMatchNumber(String matchNumber) { 
+        this.matchNumber = matchNumber;    
+    }
```
Comment (MthreeJohnnyK) [inline app/src/main/java/com/example/myapplication/MatchData.java:76]: TeamNumber should be later stored as integer if there are no teams with leading zeroes. MatchNumber should later be stored as integer no matter what. ;)

File: data processing/file_joiner.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Read the contents of each ".txt" file and append to the merged content
+for json_file in json_files:
+    file_path = os.path.join(directory_path, json_file)
+    with open(file_path, 'r', encoding='utf-8') as in_file:
+        merged_content += separator
+        separator = ','
+        merged_content += in_file.read()
+
+# Close the array
+merged_content += ']'
```
Comment (MthreeJohnnyK) [inline data processing/file_joiner.py:24]: What does this array declaration do?

**Comment by MthreeJohnnyK** [comment]
> _ No description provided. _

What is this PR supposed to do?

File: app/src/main/java/com/example/myapplication/MatchData.java (hunk 1 above)
Comment (ConfettiRaindrop) [reply app/src/main/java/com/example/myapplication/MatchData.java:76]: In the case of its usage within the scouting app, this technically isn't the case since we're only using this to match the data with the team and to place it as text on the screen, and not for any numeric operations so there aren't issues with team number being stored as a string

File: data processing/file_joiner.py (hunk 2 above)
Comment (ConfettiRaindrop) [reply data processing/file_joiner.py:24]: The merged file of all of the JSON files stores them in a list of JSON's, so the array is for the files, if you want to see what it looks like once it's been run the merged files are all in the google drive in the scouting folder

File: app/src/main/java/com/example/myapplication/csv_to_json.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import csv, json, datetime
+
```
Comment (MthreeJohnnyK) [inline app/src/main/java/com/example/myapplication/csv_to_json.py]: What does this merge do? It just says "checking in".

**Review by MthreeJohnnyK — CHANGES_REQUESTED** [review]

File: data processing/note_obtainer.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+def find_nth_occurrence_1(s, sub, n):
+	count = s.count(sub)
+	if count < n:
+		return -1
+	else:
+		index = -1
+		for i in range(n):
+			index = s.find(sub, index+1)
+		return index
+
```
Comment (MthreeJohnnyK) [inline data processing/note_obtainer.py:17]: What does this function do?

Comment (MthreeJohnnyK) [reply data processing/note_obtainer.py:17]: Add a #comment maybe?

File: data processing/csv_to_json.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        else:
+            current_record = {}
+            for i in range(len(column_names)):
+                current_record[column_names[i]] = row[i]
+            all_matches.append(current_record)
+    total_time = datetime.datetime.hour*1000 + datetime.datetime.minute*100 + datetime.datetime.second + float(1/datetime.datetime.microsecond) - current_time
+
+
+#print(all_matches)
+
```
Comment (MthreeJohnnyK) [inline data processing/csv_to_json.py:29]: Why is this print statement commented out?

File: data processing/csv_to_json.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import csv, json, datetime
+
+# Specify the directory path
+csv_file_name = r'C:\Users\camil\Robotics\JoinedData\PNCMP2024-MatchExport-Q100.csv' #TODO:check if file name need changing later
+json_file_name = r'C:\Users\camil\Robotics\JoinedData\DCMPQualMatches.json'
+
+all_matches = []
+total_time = None
+current_time = datetime.datetime.hour*1000 + datetime.datetime.minute*100 + datetime.datetime.second + float(1/datetime.datetime.microsecond)
+
```
Comment (MthreeJohnnyK) [inline data processing/csv_to_json.py:10]: What are these variables and what do they do?

File: data processing/note_obtainer.py (hunk 4 above)
Comment (MthreeJohnnyK) [reply data processing/note_obtainer.py:17]: Is it even called?

File: data processing/file_joiner.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+# Separator starts empty for first file
+separator = ''
+
+# Read the contents of each ".txt" file and append to the merged content
+for json_file in json_files:
+    file_path = os.path.join(directory_path, json_file)
+    with open(file_path, 'r', encoding='utf-8') as in_file:
+        merged_content += separator
+        separator = ','
+        merged_content += in_file.read()
```
Comment (MthreeJohnnyK) [inline data processing/file_joiner.py:21]: Better written as:
`merged_content += in_file.read() + ','`
Remove the separator on the last iteration and remove the separator variable.

**Review by MthreeJohnnyK — CHANGES_REQUESTED** [review]

File: data processing/csv_to_json.py (hunk 5 above)
Comment (MthreeJohnnyK) [reply data processing/csv_to_json.py:29]: Remove?

File: data processing/csv_to_json.py (hunk 6 above)
Comment (ConfettiRaindrop) [reply data processing/csv_to_json.py:10]: alll_matches is the array that's used to hold all of the data for the JSON files
total_time and current_time is set there to be able to figure out the total time it takes for this code to run

File: data processing/csv_to_json.py (hunk 5 above)
Comment (ConfettiRaindrop) [reply data processing/csv_to_json.py:29]: great catch, it was for testing, but since this code is done it should have been deleted

File: data processing/csv_to_json.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import csv, json, datetime
+
+# Specify the directory path
+csv_file_name = r'C:\Users\camil\Robotics\JoinedData\PNCMP2024-MatchExport-Q100.csv' #TODO:check if file name need changing later
+json_file_name = r'C:\Users\camil\Robotics\JoinedData\DCMPQualMatches.json'
```
Comment (MthreeJohnnyK) [inline data processing/csv_to_json.py:5]: Why are these variables only referenced once?

Comment (MthreeJohnnyK) [reply data processing/csv_to_json.py:5]: Why is there still a "to do"?

File: data processing/csv_to_json.py (hunk 6 above)
Comment (MthreeJohnnyK) [reply data processing/csv_to_json.py:10]: There are 3600 seconds in an hour and 60 seconds in a minute.

File: data processing/csv_to_json.py (hunk 8 above)
Comment (ConfettiRaindrop) [reply data processing/csv_to_json.py:5]: because there shouldn't be one ╰(*°▽°*)╯

File: app/src/main/java/com/example/myapplication/MatchData.java [hunk 9]
```diff
@@ -59,28 +59,46 @@ public String returnAllData() {
     //Getters and Setters
```
Comment (MthreeJohnnyK) [inline app/src/main/java/com/example/myapplication/MatchData.java:59]: Why is this here?

File: app/src/main/java/com/example/myapplication/MatchData.java [hunk 10]
```diff
@@ -59,28 +59,46 @@ public String returnAllData() {
...
+        return teamNumber;    
+    
+    }
+    public void setTeamNumber(String teamNumber) { 
+        this.teamNumber = teamNumber;   
+    }
+    
+    public String getMatchNumber() {    
+        return matchNumber;    
+    }
```
Comment (MthreeJohnnyK) [inline app/src/main/java/com/example/myapplication/MatchData.java:72]: Why is the match number a string?

File: app/src/main/java/com/example/myapplication/MatchData.java (hunk 9 above)
Comment (ConfettiRaindrop) [reply app/src/main/java/com/example/myapplication/MatchData.java:59]: it creates a basic description of what this part of the file is about

File: app/src/main/java/com/example/myapplication/MatchData.java (hunk 10 above)
Comment (ConfettiRaindrop) [reply app/src/main/java/com/example/myapplication/MatchData.java:72]: because we receive it as a string and have no reason to turn it into a number since it's only used for data output, no numerical operations

File: app/src/main/java/com/example/myapplication/MatchData.java [hunk 11]
```diff
@@ -89,56 +107,87 @@ public void setSpeakerAuto(int speakerAuto) {
...
+
     public void setAmpTeleop(int ampTeleop) {
         this.ampTeleop = ampTeleop;
     }
 
     public int getMissedNotes() {
         return missedNotes;
     }
+
     public void setMissedNotes(int missedNotes) {   this.missedNotes = missedNotes;   }
```
Comment (MthreeJohnnyK) [inline app/src/main/java/com/example/myapplication/MatchData.java:136]: Fix spacing here?

Comment (ConfettiRaindrop) [reply app/src/main/java/com/example/myapplication/MatchData.java:136]: nice catch, that's the inconsistency issue that was floating about

File: build.gradle.kts [hunk 12]
```diff
@@ -1,6 +1,6 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id("com.android.application") version "8.3.0" apply false
+    id("com.android.application") version "8.3.1" apply false
```
Comment (MthreeJohnnyK) [inline build.gradle.kts:3]: Why is a gradle file committed?

File: tool/a.exe
Comment (MthreeJohnnyK) [inline tool/a.exe:1]: Why is it named "a"? Why is the folder named "tool"? What is this supposed to do?

Comment (MthreeJohnnyK) [reply tool/a.exe:1]: And why is a binary file committed?

File: build.gradle.kts (hunk 12 above)
Comment (ConfettiRaindrop) [reply build.gradle.kts:3]: it really shouldn't be, this is something that we should probably add to the gitignore file later

File: app/src/main/java/com/example/myapplication/MatchData.java [hunk 13]
```diff
@@ -89,56 +107,87 @@ public void setSpeakerAuto(int speakerAuto) {
...
-    public void setTrapSucess(int trapSucess) {    this.trapSucess = trapSucess;    }
+    public int getTrapSucess() {     
+        return trapSucess;    
+    }
 
-    public int getTrapFail() {     return trapFail;    }
-    public void setTrapFail(int trapFail) {    this.trapFail = trapFail;    }
+    public void setTrapSucess(int trapSucess) {    
+        this.trapSucess = trapSucess;    
+    }
```
Comment (MthreeJohnnyK) [inline app/src/main/java/com/example/myapplication/MatchData.java:169]: Typo in "sucess".

File: tool/a.exe
Comment (ConfettiRaindrop) [reply tool/a.exe:1]: it really shouldn't  -> outside of the file name, you typically don't want executable files unless they have a valid reason to be there. 
However in terms of what this actually does, if you run it, it prints "hello world" to terminal

File: app/src/main/java/com/example/myapplication/MatchData.java [hunk 14]
```diff
@@ -59,28 +59,46 @@ public String returnAllData() {
...
+        this.workingAuto = workingAuto;    
+    }
     public void setMoveOutOfZone(boolean moveOutOfZone) {
         this.moveOutOfZone = moveOutOfZone;
     }
-    public boolean getMoveOutOfZone() {return moveOutOfZone;}
+
+    public boolean getMoveOutOfZone() {
+        return moveOutOfZone;
+    }
```
Comment (MthreeJohnnyK) [inline app/src/main/java/com/example/myapplication/MatchData.java:88]: Swap order for consistency.

File: tool/a.exe
Comment (MthreeJohnnyK) [inline tool/a.exe:1]: Does it do anything except opening the terminal and print something?

File: app/src/main/java/com/example/myapplication/MatchData.java (hunk 14 above)
Comment (ConfettiRaindrop) [reply app/src/main/java/com/example/myapplication/MatchData.java:88]: yep, that's one of the unnoticed issues, but something that I do want to you do is to clarify what should be swapped here since this doesn't explain what should be swapped and leaves a lot up to interpretation which you don't want when you're pointing out problems @MthreeJohnnyK

Comment (MthreeJohnnyK) [reply app/src/main/java/com/example/myapplication/MatchData.java:88]: Ok. Yeah, every time the order is get + set, but this one its set + get.

File: data processing/file_joiner.py [hunk 15]
```diff
@@ -0,0 +1,31 @@
+import os
+
+# Specify the directory path
+directory_path = r'C:\Users\camil\Documents\ScoutingData'
+
+# Get a list of all ".txt" files in the directory
```
Comment (MthreeJohnnyK) [inline data processing/file_joiner.py:6]: Inaccurate comment? Aren't you getting .json files?

File: data processing/csv_to_json.py [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+
+all_matches = []
+total_time = None
+current_time = datetime.datetime.hour*1000 + datetime.datetime.minute*100 + datetime.datetime.second + float(1/datetime.datetime.microsecond)
```
Comment (MthreeJohnnyK) [inline data processing/csv_to_json.py:9]: This is called 3 times throughout your code...
